lib/streamlit/elements/skeleton.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Optional, cast
+
+from streamlit.proto.Skeleton_pb2 import Skeleton as SkeletonProto
+from streamlit.runtime.metrics_util import gather_metrics
+
+if TYPE_CHECKING:
+    from streamlit.delta_generator import DeltaGenerator
+
+
+class SkeletonMixin:
+    @gather_metrics("skeleton")
+    def skeleton(
+        self, height: Optional[int] = None, *, width: Optional[float] = None
+    ) -> "DeltaGenerator":
+        """Insert a single-element container which displays a "skeleton" animation.
+
+        Inserts a container into your app that can be used to hold a single element.
+        This allows you to, for example, remove elements at any point, or replace
+        several elements at once (using a child multi-element container).
+
+        To insert/replace/clear an element on the returned container, you can
+        use "with" notation or just call methods directly on the returned object.
+        See some of the examples below.
+
+        Parameters
+        ----------
+        height (optional): int
+            An integer representing the height in CSS points of the adjustable-height block.
```

Comment:
We already have descriptions for `height` for elements like `st.dataframe` and `st.container` and can align the description text a bit more to that:

```suggestion
            Desired height of the skeleton expressed in pixels. If ``None`` (default), a default height is used.
```

Reply:
@LukasMasuch based on the other protos that include dimensional elements, the defacto standard is to use "css points", not pixels. That's why the component uses "pt" instead of "px".

Which do we prefer here? Pixels or points?

Reply:
I think we prefer pixels. At least thats what we have for all the other elements that have a `width`, e.g.: [st.dataframe](https://docs.streamlit.io/library/api-reference/data/st.dataframe), [st.line_chart](https://docs.streamlit.io/library/api-reference/charts/st.line_chart), and [st.image](https://docs.streamlit.io/library/api-reference/media/st.image).